src/borg/key.py

```diff
@@ -171,7 +171,7 @@ def id_hash(self, data):
 
     def encrypt(self, chunk):
         chunk = self.compress(chunk)
-        self.nonce_manager.ensure_reservation(num_aes_blocks(len(chunk.data)))
+        self.nonce_manager.ensure_reservation(self.cipher.block_count(len(chunk.data)))
```

Comment:
What we care about here is the amount of nonce space used. Which i think is not block_count in the new generalized setting.